### ISSUE TITLE
Notification Email "Notes" not accurate for Reviewer User Group 2 and Publisher User Group

### ISSUE THREAD

**Issue by riusi88** [engram-design/Workflow#147]
Really appreciate this plugin as it's been a game changer for us. We did run into an issue we're hoping you can investigate. 

My Workflow setup is as follows: Editor Group > Reviewer Group 1 > Reviewer Group 2 > Publisher Group. All Groups setup to receive email notifications. 

When an Editor submits a draft for review (with notes), the notes that appear in the notification email for Reviewer Group 1 include the correct Editor Group 1 notes. However, when Reviewer Group 1 approves a draft (with notes), the notification email that Reviewer Group 2 receives includes Editor Group notes and not Reviewer Group 1 notes. Similarly, the Publisher Group notifications also include Editor Group notes, and not Reviewer Group 2 notes. This presents an issue for Reviewer Group 2 and Publishers as they are provided the incorrect notes when they begin approving the draft via the email chain. It's not a site breaking bug, but we're hoping to avoid explaining this issue to our reviewers.

What's odd is that I enabled the Editor Group to be CC'd on the Reviewer email notifications under "Editor Notifications - Additional Options", and the Editor Group correctly receives the Reviewer Group 1 and 2 notes when those notifications go out. 
 

**Steps to reproduce**
1. Setup is as follows: Editor Group > Reviewer Group 1 > Reviewer Group 2 > Publisher Group. All Groups setup to receive email notifications. 
2. Editor creates draft with notes
3. Reviewer Group 1 receives email with Editor Group notes, clicks the entry edit link and approves draft and adds notes
4. Reviewer Group 2 receives email with Editor Group notes (not Reviewer Group 1 notes)

**Additional info**
- Plugin version: dev-craft-3 as 1.7.2
- - I submitted this issue https://github.com/verbb/workflow/issues/145 so have this version of the plugin currently installed. 
- Craft version: Craft Pro 3.7.37
- Multi-site: No

**Comment by engram-design** [comment]
So the issue is that the emails sent to the next reviewer is exactly the same as the email from Editor to Reviewer 1, and what's more, any further emails all the way to Publishers won't contain the full stream of notes from Editor to Publisher.

So firstly, we should probably introduce another email to send out to reviewers, to say that something is available for review.

But the other thing to consider is whether we should show the entire review processes worth of notes. So for instance, the email a Publisher gets might look like:

- Editor Notes: Some notes
- Reviewer 1 Notes: More notes
- Reviewer 2 Notes: Even more notes

The only issue I see with this, would be the emails back to the editor. These can be sent every time an entry is approved for each review (see "Reviewer Approval Notifications"), or at the very end, when the Publisher approves. I'm not sure is we should show all the reviewer/publishers notes to the editor. Maybe these notes are only for the next approver level, and not for the original editor? I'm also not sure what would be more helpful...

**Comment by engram-design** [comment]
The original issue should be fixed for the next release. To get the fix early, change your `verbb/workflow` requirement in `composer.json` to:

```
"require": {
  "verbb/workflow": "dev-craft-3 as 1.7.2",
  "...": "..."
}
```

Then run `composer update`.

But keen to know if it'd be worthwhile having the full list of notes shown in email notifications.

**Comment by riusi88** [comment]
hey @engram-design I appreciate the quick turn around on this. You bring up some valid questions regarding what information would be useful to include in the Publisher and Editor notifications. My opinion on the matter would be the following:

Publishers:
- In some cases it would be useful for publishers to know the entire approval and review notes chain upon receiving the notification email. In most scenarios Publishers should be established as "owners" of the content distribution chain. Therefore providing them vision across the entire review process would be helpful for them to determine the entire scope of "who, what, where and why/how" a specific post landed in their lap for approval and publishing. 
- I imagine if we throw rejections into the mix, this might get more complicated. However, the job of a reviewer ought to be someone who's responsibility is handling rejections and making the Publishers job less arduous. So I don't think rejection notes would be important to include in the chain, as it might produce more questions for those higher up in the approvals process. 
- In truth my point above can also be applied to approval notes. Is it in the best interest of "good process" to increase the complexity of the notes a Publisher receives. Should they only see the last set of reviewer notes since it's the reviewers job to "care" about editor and previous editor notes? For well oiled efficient teams maybe the answer is yes -- For teams that require more oversite through the review phases, perhaps it is important for a Publisher to be aware of the loop, and not just the end of it. 

Editors:
- With regards to editors things get a bit more complicated. If we consider companies that would use workflow to prevent editors from having publisher access then I imagine in most cases they may also not want editors to be explicitly sent review notes that occur up the chain. 
- Alternatively, companies can use this as an editor teaching tool. Providing them the entire chain of notes may work towards teaching them company or publishing best practices. 

Suffice it to say I think the way it works now is completely fine, but perhaps having this kind of thing as an option in the future could certainly be helpful.

**Comment by engram-design** [comment]
Thanks for the feedback and thoughts - very useful! I'll continue to give it some more thought, as there's certainly some scenarios to consider here.

My immediate thought for down the track is to provide an array of notes when you could loop through in emails. At least that way it'd be up to you and your needs on what needs to be shown:

#### Summary of notes for all submissions
`{{ submission.getNotesSummary() }}`

> **Editor:** "Might need some help with..."
> **Reviewer 1 (Rejected):** "This was a great start, but..."
> **Reviewer 1 (Accepted):** "Much better, thanks!"
> **Reviewer 2 (Accepted):** "Love it - no issues from me."
> **Publisher (Accepted):** "Great work team!"

This would also change to reflect the currently available notes. So for Reviewer 2, you'd only see the first 2 note items, because the latter 2 hadn't been input yet.

#### Loop through notes

```
{% for note in submission.getNotes() %}
    {# Show notes to everyone, unless this is going to an editor #}
    {% if submission.notificationType == 'editor' %}
        {# Do nothing #}
    {% else %}
        {# If this was a publisher note, don't show #}
        {% if note.type != 'publisher' %}
            {{ note.note }}
        {% endif %}
    {% endif %}
{% endfor %}
```

Lot's of options here, including being able to tell who the recipient is, and what type of note it is.

**Comment by engram-design** [comment]
This should be solved in [2.0.0](https://github.com/verbb/workflow/releases/tag/2.0.0) thanks to a big refactor with reviews, this should now be possible. Refer to [blog post](https://verbb.io/blog/workflow-2) and [upgrade docs](https://verbb.io/craft-plugins/workflow/docs/get-started/upgrading-from-v1).
